src/Microsoft.HttpRepl.IntegrationTests/Mocks/MockHttpMessageHandler.cs

```diff
@@ -0,0 +1,24 @@
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.HttpRepl.IntegrationTests.Mocks
+{
+    public class MockHttpMessageHandler : HttpMessageHandler
+    {
+        private HttpResponseMessage response;
+
+        public MockHttpMessageHandler(HttpResponseMessage response)
+        {
+            this.response = response;
+        }
+
+        protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            var responseTask = new TaskCompletionSource<HttpResponseMessage>();
```

Comment:
how is this body different from `Task.FromResult(response)`?

Reply:
Nope, it's the same. Updated here:
https://github.com/aspnet/HttpRepl/commit/383ed286ac8470d2a34f7a4f9d2d36857aec4685